src/test/java/org/broadinstitute/hellbender/engine/FeatureInputUnitTest.java

```diff
@@ -56,32 +66,107 @@ public void testInvalidFeatureArgumentValue( final String invalidFeatureArgument
     public void testNoFeatureNameSpecified(final String validFileOnlyFeatureArgumentValue) {
         FeatureInput<Feature> featureInput = new FeatureInput<>(validFileOnlyFeatureArgumentValue);   //"myName,key1=value,myFile"
 
-        Assert.assertEquals(featureInput.getFeatureFile(), new File(validFileOnlyFeatureArgumentValue), "Wrong File in FeatureInput");
+        Assert.assertEquals(featureInput.getFeaturePath(), validFileOnlyFeatureArgumentValue, "Wrong File in FeatureInput");
         // Name should default to the absolute path of the File when no name is specified
         Assert.assertEquals(featureInput.getName(), new File(validFileOnlyFeatureArgumentValue).getAbsolutePath(), "Wrong default name in FeatureInput");
     }
 
+    @DataProvider
+    public Object[][] getRawArgumentsWithURI(){
+        return new Object[][]{
+                {"name:file://myFile", "name", "file://myFile"},
+                {"name:gendb://myJsons", "name", "gendb://myJsons"},
+                {"name"}
+        };
+    }
+
+
+
+    @Test
+    public void testGenDbPathAndName(){
+        final FeatureInput<VariantContext> gendbInput = new FeatureInput<>("gendb://myJsons");
+
+        Assert.assertEquals(gendbInput.getFeaturePath(), "gendb://myJsons", "wrong featurePath");
+        Assert.assertEquals(gendbInput.getName(), "gendb://" + new File("myJsons").getAbsolutePath());
+    }
+
+    @DataProvider
+    public Object[][] getFeatureInputs(){
+
+        return new Object[][] {
+                {new FeatureInput<VariantContext>(FeatureInput.GENOMIC_DB_URI_SCHEME + jsonPath  ), true, false},
+                {setFeatureTypeToVariant(new FeatureInput<>(vcfWithIndex)), true, true},
+                {setFeatureTypeToVariant(new FeatureInput<>(vcfWithoutIndex)), false, false},
+                {setFeatureTypeToVariant(new FeatureInput<>("name", Collections.emptyMap(), vcfWithIndex)), true, true},
+                {setFeatureTypeToVariant(new FeatureInput<>("name", Collections.emptyMap(), new File(vcfWithIndex))), true, true},
+                {new FeatureInput<>("name", new VCFCodec(), new File(vcfWithIndex)), true, true},
+        };
+    }
+
+    private static FeatureInput<VariantContext> setFeatureTypeToVariant(FeatureInput<VariantContext> stringStringConstructed) {
+        stringStringConstructed.setFeatureType(VariantContext.class);
+        return stringStringConstructed;
+    }
+
+    private void assertProducesFeatureReader(FeatureInput<?> input) throws IOException {
+        Assert.assertNotNull(input.getFeatureReader().iterator().next());
+    }
+
+    @Test(dataProvider = "getFeatureInputs")
+    public void testGetFeatureReader(FeatureInput<VariantContext> input, boolean ignored, boolean alsoIgnored) throws IOException {
+        if(FeatureInput.isGenomeDBPath(input.getFeaturePath())) {
+            GenomicsDBIntegrationTest.runOnGenomicsDBArray(() -> {
```

Comment:
It's bad practice to call into methods from other test classes -- if necessary, create a common utility in `utils.test`